src/runlib.rs

```diff
@@ -22,11 +22,45 @@ use crate::{Error, Result};
 pub fn record_artifact(
     path: &str,
     hash_algorithms: &[HashAlgorithm],
+    lstrip_paths: Option<&[&str]>,
 ) -> Result<(VirtualTargetPath, TargetDescription)> {
     let file = File::open(path)?;
     let mut reader = BufReader::new(file);
     let (_length, hashes) = crypto::calculate_hashes(&mut reader, hash_algorithms)?;
-    Ok((VirtualTargetPath::new(String::from(path))?, hashes))
+    let lstripped_path = apply_left_strip(path, lstrip_paths)?;
+    Ok((VirtualTargetPath::new(lstripped_path)?, hashes))
+}
+
+/// Given an artifact path in `&str` format, left strip path for given artifact based an optional array of `lstrip_paths` provided,
+/// returning the stripped file path in String format wrapped in `Result`.
+fn apply_left_strip(path: &str, lstrip_paths: Option<&[&str]>) -> Result<String> {
+    // If lstrip_paths is None, skip strip.
+    // Else, check if path starts with any given lstrip paths and strip
+    if let None = lstrip_paths {
+        return Ok(String::from(path));
+    }
+    let l_paths = lstrip_paths.unwrap();
+    let mut stripped_path = path;
+    let mut find_prefix = "";
+    for l_path in l_paths.iter() {
+        if !path.starts_with(l_path) {
+            continue;
+        }
+        if !find_prefix.is_empty() && find_prefix.len() >= l_path.len() {
+            continue;
+        }
```

Comment:
Do I understand this guard properly? wouldn't this be better if we inverted and turned into a `break` ?

Reply:
If there is two or more substitution, we cannont break immediately break it.